handlers/deploy.go

```diff
@@ -344,38 +344,62 @@ func createResources(request requests.CreateFunctionRequest) (*apiv1.ResourceReq
 		Requests: apiv1.ResourceList{},
 	}
 
-	// Set Memory limits
-	if request.Limits != nil && len(request.Limits.Memory) > 0 {
-		qty, err := resource.ParseQuantity(request.Limits.Memory)
-		if err != nil {
-			return resources, err
+	if request.Limits != nil {
+		if len(request.Limits.Memory) > 0 {
+			// Set Memory limits
+			qty, err := resource.ParseQuantity(request.Limits.Memory)
+			if err != nil {
+				return resources, err
+			}
+			resources.Limits[apiv1.ResourceMemory] = qty
 		}
-		resources.Limits[apiv1.ResourceMemory] = qty
-	}
-
-	if request.Requests != nil && len(request.Requests.Memory) > 0 {
-		qty, err := resource.ParseQuantity(request.Requests.Memory)
-		if err != nil {
-			return resources, err
+		if len(request.Limits.CPU) > 0 {
+			// Set CPU limits
+			qty, err := resource.ParseQuantity(request.Limits.CPU)
+			if err != nil {
+				return resources, err
+			}
+			resources.Limits[apiv1.ResourceCPU] = qty
 		}
-		resources.Requests[apiv1.ResourceMemory] = qty
-	}
-
-	// Set CPU limits
-	if request.Limits != nil && len(request.Limits.CPU) > 0 {
-		qty, err := resource.ParseQuantity(request.Limits.CPU)
-		if err != nil {
-			return resources, err
+		if len(request.Limits.Others) > 0 {
+			// Set other extended resources
+			for extresource, value := range request.Limits.Others {
+				qty, err := resource.ParseQuantity(value)
+				if err != nil {
+					return resources, err
+				}
+				resources.Limits[apiv1.ResourceName(extresource)] = qty
+			}
 		}
-		resources.Limits[apiv1.ResourceCPU] = qty
 	}
 
-	if request.Requests != nil && len(request.Requests.CPU) > 0 {
-		qty, err := resource.ParseQuantity(request.Requests.CPU)
-		if err != nil {
-			return resources, err
+	if request.Requests != nil {
+		if len(request.Requests.Memory) > 0 {
+			// Set Memory limits
+			qty, err := resource.ParseQuantity(request.Requests.Memory)
+			if err != nil {
+				return resources, err
+			}
+			resources.Requests[apiv1.ResourceMemory] = qty
+		}
+		if len(request.Requests.CPU) > 0 {
+			// Set CPU limits
+			qty, err := resource.ParseQuantity(request.Requests.CPU)
+			if err != nil {
+				return resources, err
+			}
+			resources.Requests[apiv1.ResourceCPU] = qty
```

Comment:
This took me a while to get to. Do we need to update the deserialization code here or is it not needed?

Reply:
No worries. Technically the patch works as-is, considering that one uses the patched faas-cli too. However I am not totally satisfied. I believe that in the long run it would be nice to specify what exactly allowed as "extended resource". We must be able to validate and sanitize the input on the netes side. For now this sanitation happens on the client side only. There my patch makes sure that only things in the form of: vendor.domain/[gpu,fpga] are allowed to go through.

Reply:
The syntax for defining limits for Intel device plugins have changed as seen in this [example YAML](https://github.com/intel/intel-device-plugins-for-kubernetes/blob/master/demo/intelgpu-job.yaml#L22). 
This patch for faas-netes and the [patch for faas-cli](https://github.com/openfaas/faas-cli/pull/488) needs a re-write.